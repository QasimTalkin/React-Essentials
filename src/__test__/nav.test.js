import {render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '../components/Organism/Nav';

describe('Nav Component renders correctly', () => {

  it ('renders correctly', () => {
    const { asFragment } = render(<Nav />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Nav bar has home and about links', () => {
    const { getByText } = render(<Nav />);
    const homeLink = getByText(/home/i);
    const aboutLink = getByText(/about/i);
    
    expect(homeLink).toBeInTheDocument();
    expect(aboutLink).toBeInTheDocument();
  });
  
  it ('contains a logo H2 ReactCoffee', () => {
    const { getByText } = render(<Nav />);
    const logo = getByText(/reactcoffee/i);
    expect(logo).toBeInTheDocument();
  });

});