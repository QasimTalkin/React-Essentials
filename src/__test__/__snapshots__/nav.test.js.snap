// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component renders correctly renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="w-full bg-white shadow"
  >
    <div
      class="justify-between px-4 mx-auto lg:max-w-7xl md:items-center md:flex md:px-8"
    >
      <div>
        <div
          class="flex items-center justify-between py-3 md:py-5 md:block"
        >
          <a
            href="/"
          >
            <h2
              class="text-2xl font-bold"
            >
              ReactCoffee
            </h2>
          </a>
          <div
            class="md:hidden"
          >
            <button
              class="p-2 text-gray-700 rounded-md outline-none focus:border-gray-400 focus:border"
            >
              <svg
                class="w-6 h-6"
                fill="none"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 12h16M4 18h16"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          class="flex-1 justify-self-center pb-3 mt-8 md:block md:pb-0 md:mt-0 hidden"
        >
          <ul
            class="items-center justify-center space-y-8 md:flex md:space-x-6 md:space-y-0"
          >
            <li
              class="text-gray-600 hover:text-blue-600"
            >
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="text-gray-600 hover:text-blue-600"
            >
              <a
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="text-gray-600 hover:text-blue-600"
            >
              <a
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
